src/Cache/CacheEngine.php

```diff
@@ -17,6 +17,7 @@
 namespace Cake\Cache;
 
 use Cake\Core\InstanceConfigTrait;
+use Cake\Http\Exception\NotImplementedException;
```

Comment:
Using a HTTP exception doesn't make sense here. You can add a new exception class under the Cache/Exception namespace which extends CakeException.

Reply:
OK, I'll try it 💪

Reply:
fixed https://github.com/cakephp/cakephp/pull/16478/commits/676c65d3048c1f6a8eaa5f9c17c65939b103f6bf